Mathlib/NumberTheory/Ostrowski.lean

```diff
@@ -319,16 +319,20 @@ lemma one_lt_of_not_bounded (notbdd : ¬ ∀ n : ℕ, f n ≤ 1) {n₀ : ℕ} (h
   · simp only [CharP.cast_eq_zero, map_zero, zero_le_one]
   have h_ineq2 (k : ℕ) (hk : 0 < k) :
       f n ≤ (n₀ * (logb n₀ n + 1)) ^ (k : ℝ)⁻¹ * k ^ (k : ℝ)⁻¹ := by
-    have h_exp : (f n ^ (k : ℝ)) ^ (k : ℝ)⁻¹ = f n := rpow_rpow_inv (by positivity) (by positivity)
     have : 0 ≤ logb n₀ n := logb_nonneg (one_lt_cast.2 hn₀) (mod_cast Nat.one_le_of_lt h₀.bot_lt)
-    rw [← mul_rpow (by positivity) (by positivity), ← h_exp, rpow_natCast, ← map_pow,
-      ← Nat.cast_pow]
-    gcongr
-    apply le_trans (h_ineq1 (one_le_pow k n h₀.bot_lt))
-    rw [mul_assoc, Nat.cast_pow, logb_pow (mod_cast h₀.bot_lt), mul_comm _ (k : ℝ),
+    calc
+    f n = (f n ^ (k : ℝ)) ^ (k : ℝ)⁻¹ := Eq.symm (rpow_rpow_inv (by positivity) (by positivity))
+    _   ≤ (n₀ * (logb n₀ (n ^ k) + 1)) ^ (k : ℝ)⁻¹ := by
+      gcongr
+      rw [rpow_natCast, ← map_pow, ← Nat.cast_pow, ← Nat.cast_pow]
+      exact (h_ineq1 (one_le_pow k n h₀.bot_lt))
```

Comment:
Can you avoid meddling inequality steps and rearrangement steps? Here this involves adding a new `_ = ... := ...` line so that the proof here becomes
```suggestion
    _   ≤ ... := by
      gcongr
      exact h_ineq1 (one_le_pow k n h₀.bot_lt)
```

Reply:
Done. the block now consists of a rearrangement, an inequality, a rearrangement, an inequality and one last rearrangement.